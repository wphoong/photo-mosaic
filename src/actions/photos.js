import uuid from "uuid";

// ADD PHOTO

export const addPhoto = (
  {
    photoLink = "",
    title = "",
    createdAt = 0,
    description = ""
  } = {}) => ({
    type: "ADD_PHOTO",
    photo: {
      id: uuid(),
      photoLink,
      title,
      createdAt,
      description
    }
  }
);

export const editPhoto = (id, updates) => ({
  type: "EDIT_PHOTO",
  id,
  updates
});

export const removePhoto = ({id} = {}) => ({
  type: "REMOVE_PHOTO",
  id
});

// export const startAddPhoto = (photoData = {}) => {
//   return (dispatch, getState) => {
//     const uid = getState().auth.uid;
//     const {
//       photoLink = "",
//       title = "",
//       description = "",
//       createdAt = 0
//     } = photoData;
//     const photo = { photoLink, title, description, createdAt };
    
//     return dispatch(addPhoto({
//       id: ref.key,
//       ...photo 
//     }));
//   };
// };