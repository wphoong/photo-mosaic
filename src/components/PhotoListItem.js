import React from "react";
import { connect } from "react-redux";
import moment from "moment";
import { Link } from "react-router-dom";

export const PhotoListItem = ({ id, photoLink, title, createdAt, description }) => (
  <div>
    <img src={photoLink} height="300" width="300"/>
    <Link to={`/edit/${id}`}>
      <h3>{title}</h3>
    </Link>
    <p>{description}</p>
    <p>Created at: {moment(createdAt).format("MMM Do, YYYY")}</p>
  </div>
);

export default connect()(PhotoListItem);

