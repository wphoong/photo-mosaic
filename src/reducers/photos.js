const photosReducerDefaultState = [];

const photosReducer = (state = photosReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_PHOTO":
      return [
        ...state,
        action.photo
      ];
    default:
      return state;
  }
};

export default photosReducer;