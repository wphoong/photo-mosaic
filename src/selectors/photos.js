import moment from "moment";

const getVisiblePhotos = (photos, { text, sortBy, startDate, endDate}) => {
  return photos.filter((photo) => {
    const createdAtMoment = moment(photos.createdAt);
    const setTextMatch = photo.title.toLowerCase().includes(text.toLowerCase());
    const startDateMatch = startDate ? startDate.isSameOrBefore(createdAtMoment, "day") : true;
    const endDateMatch = endDate ? endDate.isSameOrAfter(createdAtMoment, "day") : true;

    return setTextMatch && startDateMatch && endDateMatch;
  }).sort((a, b) => {
      if (sortBy === "descend") {
        return b.createdAt - a.createdAt;
      } else {
        return a.createdAt - b.createdAt;
      }
    }
  );
};

export default getVisiblePhotos;
