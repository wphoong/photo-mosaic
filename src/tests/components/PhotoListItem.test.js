import React from "react";
import { shallow } from "enzyme";
import { PhotoListItem } from "../../components/PhotoListItem.js";
import photos from "../fixtures/photos.js";

test("should render photolist item correctly", () => {
  const wrapper = shallow(<PhotoListItem key={photos[0].id} {...photos[0]} />);
  expect(wrapper).toMatchSnapshot();
});