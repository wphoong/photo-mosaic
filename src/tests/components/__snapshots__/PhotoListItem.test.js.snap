// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render photolist item correctly 1`] = `
<div>
  <img
    height="300"
    src="ez"
    width="300"
  />
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      test1
    </h3>
  </Link>
  <p>
    descript1
  </p>
  <p>
    Created at: 
    Dec 31st, 1969
  </p>
</div>
`;
